coord/src/module.c

```diff
@@ -928,6 +933,8 @@ static void proccessKNNSearchReply(MRReply *arr, searchReducerCtx *rCtx, RedisMo
         heap_offerx(reduceSpecialCaseCtx->knn.pq, resWrapper);
         rCtx->cachedResult = largest->result;
         rm_free(largest);
+      } else {
+        rCtx->cachedResult = res;
```

Comment:
Is this expected to be part of the merge?

Reply:
Yes. It is a significant enough leak that it was separated out to its own PR. But part of the larger effort to eliminate leaks.